base/abstractdict.jl

```diff
@@ -474,14 +474,16 @@ function isequal(l::AbstractDict, r::AbstractDict)
 end
 
 function ==(l::AbstractDict, r::AbstractDict)
-    l === r && return true
+    if l === r
+        return any(ismissing, values(l)) ? missing : true
+    end
     if isa(l,IdDict) != isa(r,IdDict)
         return false
     end
     length(l) != length(r) && return false
     anymissing = false
     for pair in l
-        isin = in(pair, r, ==)
+        isin = in(pair, r)
```

Comment:
Should the key be compared with `isequal` and value compared with `==`?

Reply:
Yes, that's what this does. I just want to phase out the third argument to `in`.

Reply:
Oh I see, thanks.

So are we using `==` everywhere for `in` except for `AbstractSet` using `isequal`?